### PR TITLE
ci: Add k8s e2e test for clh

### PR DESCRIPTION
**Pull request by jcvenegas**
Add k8s e2e test for clh.

Fixes: #2417

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

**Comment by grahamwhaley** [comment]
what's the plan here @jcvenegas - should we DNM this whilst it gets fixed up?

File: Makefile [hunk 1]
```diff
@@ -230,6 +230,12 @@ kubernetes:
 	bash -f .ci/install_bats.sh
 	bash -f integration/kubernetes/run_kubernetes_tests.sh
 
+kubernetes-e2e:
+	cd "${GOPATH}/src/github.com/kata-containers/tests/integration/kubernetes/e2e_conformance" &&\
```
Comment (chavafg) [inline Makefile:234]: does this needs to be full path? or can it be: `cd integration/kubernetes/e2e_conformance`

File: integration/kubernetes/e2e_conformance/run.sh [hunk 2]
```diff
@@ -66,48 +69,83 @@ run_sonobuoy() {
...
+			die "Found failed tests in end-to-end k8s test"
+		fi
+		local jobs_file="${SCRIPT_PATH}/e2e_k8s_jobs.yaml"
+		local expected_passed_query="jobs.${CI_JOB:-}.passed"
+		local expected_passed=$("${GOPATH}/bin/yq" read "${jobs_file}" "${expected_passed_query}")
+		if [ "${expected_passed}" != "" ];then
+			passed_query='.plugins | [ .[]."result-counts".passed] | add'
+			passed=$(sonobuoy status --json | jq "${passed_query}")
+			if [ "${expected_passed}" != "${passed}" ];then
+				die "expected ${expected_passed} test are passing, but only ${passed} passed"
```
Comment (chavafg) [inline integration/kubernetes/e2e_conformance/run.sh:124]: rewording: `expected  ${expected_passed} tests to pass, but only  ${passed} passed`

File: integration/kubernetes/e2e_conformance/run.sh [hunk 3]
```diff
@@ -66,48 +69,83 @@ run_sonobuoy() {
...
+			else
+				info "All test passed as expected ${passed}"
```
Comment (chavafg) [inline integration/kubernetes/e2e_conformance/run.sh:126]: `All ${passed} tests passed as expected`?

File: integration/kubernetes/e2e_conformance/e2e_k8s_jobs.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) 2021 Intel Corporation
```
Comment (chavafg) [inline integration/kubernetes/e2e_conformance/e2e_k8s_jobs.yaml:2]: 2020?

File: integration/kubernetes/e2e_conformance/setup.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# Entry point to start a K8S cluster for end-to-end testing
+# If cluster is already running with Kata runtime class use run.sh
```
Comment (chavafg) [inline integration/kubernetes/e2e_conformance/setup.sh:9]: If cluster is already running with Kata RuntimeClass, skip this script and use `run.sh`?

File: integration/kubernetes/e2e_conformance/e2e_k8s_jobs.yaml (hunk 4 above)
Comment (jcvenegas) [reply integration/kubernetes/e2e_conformance/e2e_k8s_jobs.yaml]: I come from the future

Comment (chavafg) [reply integration/kubernetes/e2e_conformance/e2e_k8s_jobs.yaml]: haha, that was my guess :P

**Pull request merged**